examples/cbindings/waku_example.c

```diff
@@ -11,22 +10,26 @@
 #include <unistd.h>
 #include <sys/syscall.h>
 
-#include "libwaku.h"
+#include "base64.h"
+#include "../../library/libwaku.h"
 
-// Keep a global string to store the waku call responses
-static NimStringDesc wakuString;
-NimStringDesc* mResp = &wakuString;
+#define WAKU_CALL(call)                                                        \
+do {                                                                           \
+  int ret = call;                                                              \
+  if (ret != 0) {                                                              \
+    printf("Failed the call to: %s. Returned code: %d\n", #call, ret);         \
+    exit(1);                                                                   \
+  }                                                                            \
+} while (0)
```

Comment:
I'm curious on why is the `do while` needed here :thinking:

Reply:
Thanks for the review!

I agree that this seems pointless but its main purpose is to make sure the macro always behaves as it is expected to behave.

We have good examples in the next:
https://hownot2code.wordpress.com/2016/12/05/do-while-0-in-macros/